test/enrollmentkey_test.go

```diff
@@ -0,0 +1,138 @@
+//go:build integration
+// +build integration
+
+package test
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/gravitl/netmaker/cli/config"
+	"github.com/gravitl/netmaker/cli/functions"
+	controller "github.com/gravitl/netmaker/controllers"
+	"github.com/gravitl/netmaker/database"
+	"github.com/gravitl/netmaker/logic"
+	"github.com/gravitl/netmaker/models"
+	"github.com/gravitl/netmaker/servercfg"
+	"github.com/stretchr/testify/assert"
+)
+
+func DBInit() {
+	database.InitializeDatabase()
+	database.DeleteAllRecords(database.USERS_TABLE_NAME)
+	database.DeleteAllRecords(database.NETWORKS_TABLE_NAME)
+	database.DeleteAllRecords(database.NETWORK_USER_TABLE_NAME)
+	database.DeleteAllRecords(database.ENROLLMENT_KEYS_TABLE_NAME)
+	// TODO rest
+}
+
+func TestHasNetworksAccessAPI(t *testing.T) {
+	// setup / teardown (TODO extract)
+	DBInit()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	go func() {
+		wg.Wait()
+		defer database.CloseDB()
+	}()
+	var err error
+	port := servercfg.GetAPIPort()
+	userPass := "bar123"
+	user := &models.User{
+		UserName: "foo",
+		Password: userPass,
+		// TODO should be handled in fixtures?
+		Networks: []string{"network-1"},
+		IsAdmin:  false,
+		Groups:   nil,
+	}
+	err = logic.CreateUser(user)
+	if err != nil {
+		t.Error("Error creating a user ", err)
+	}
+	userConfig := config.Context{
+		Endpoint: "http://localhost:" + port,
+		Username: user.UserName,
+		Password: userPass,
+	}
+	adminConfig := userConfig
+	adminConfig.MasterKey = "foo123"
+	adminConfigBad := userConfig
+	adminConfigBad.MasterKey = "wrongpass"
+	adminConfigBad.Password = "wrongpass"
+	config.SetContext("user-ctx-1", userConfig)
+	config.SetContext("admin-ctx-1", adminConfig)
+	config.SetContext("admin-ctx-2", adminConfigBad)
+	config.SetCurrentContext("user-ctx-1")
```

Comment:
what is the role of `SetContext` and `SetCurrentContext` here?

Reply:
It sets the configs (as in `nmctl context list`). Ive added some comments to those. Those arent golang's context.

Reply:
But isn't these tests for apis?
why is context set here for cli?

Reply:
It uses the httpclient from `netmaker/cli/functions` (via `GetEnrollmentKeys` in this case), which is a good thing, although that httpclient is too limited ATM.